### PR TITLE
Add ability to save an array of models

### PR DESCRIPTION
**Pull request by twof**
Adds `.save(on:)` to collections of `Model`s so users can do something like

```swift
let todos = [
    Todo(userId: newUser.id!, title: "Do the thing"),
    Todo(userId: newUser.id!, title: "Do the other thing")
]
todos.save(on: req)
```

### CODE REVIEW

File: Sources/Fluent/Model/Collection+Model.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            return self.map { $0.update(on: conn) }.flatten(on: conn)
+        }
+    }
+    
+    /// Creates a collection of models in Database
+    ///
+    /// - Parameter conn: A means to connect to a database
+    /// - Returns: A future array of the created models
+    func create(on conn: Element.Database.Connection) -> Future<[Element]> {
+        let databaseIdentifier = Element.defaultDatabase!
```
Comment (twof) [inline Sources/Fluent/Model/Collection+Model.swift]: I don't know in what situations `defaultDatabase` will be `nil`, so I'm not sure how folks want it handled.

**Comment by flashspys** [comment]
Why is that still not merged after almost a year?

**Comment by twof** [comment]
@tanner0101 You asked about the transaction usage in the discord. To answer your question. I opted for transaction for performance reasons, but I think the concerns you expressed are legitimate, so I'll de-transact.

File: Sources/Fluent/Model/Collection+Model.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import Async
+
+extension Collection where Element: Model, Element.Database: TransactionSupporting {
+    /// Saves a collection of models to Database
+    ///
+    /// - Parameter conn: A means to connect to a database
+    /// - Returns: A future array of the saved models
+    func save(on conn: Element.Database.Connection) -> Future<[Element]> {
+        return self.map { $0.save(on: dbConn) }.flatten(on: dbConn)
```
Comment (code28) [inline Sources/Fluent/Model/Collection+Model.swift]: After removing the transaction part, you missed the renaming: `dbConn` should bei `conn`.

```suggestion
        return self.map { $0.save(on: conn) }.flatten(on: conn)
```

File: Sources/Fluent/Model/Collection+Model.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    public func update(on conn: DatabaseConnectable) -> Future<[Element]> {
+        return self.map { $0.update(on: conn) }.flatten(on: conn)
+    }
+    
+    /// Creates a collection of models in Database
+    ///
+    /// - Parameter conn: A means to connect to a database
+    /// - Returns: A future array of the created models
+    func create(on conn: Element.Database.Connection) -> Future<[Element]> {
+        return self.map { $0.create(on: dbConn) }.flatten(on: dbConn)
```
Comment (code28) [inline Sources/Fluent/Model/Collection+Model.swift]: Same here: `dbConn` to `conn`.

```suggestion
        return self.map { $0.create(on: conn) }.flatten(on: conn)
```

File: Sources/Fluent/Model/Collection+Model.swift [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    }
+    
+    /// Deletes a collection of models from Database
+    ///
+    /// - Parameter conn: A means to connect to a database
+    /// - Returns: A future array of the deleted models
+    func delete(on conn: Element.Database.Connection) -> Future<[Element]> {
+        return element.delete(on: dbConn).transform(to: element)
```
Comment (code28) [inline Sources/Fluent/Model/Collection+Model.swift]: Here you deleted too much as well, I think:

```suggestion
        return self.map { element in
            return element.delete(on: conn).transform(to: element)
        }.flatten(on: conn)
```

**Comment by tanner0101** [comment]
This has been added in Fluent 4.

**Comment by grennis** [comment]
@twof @tanner0101 In this PR to add ability to save models, only the `update` function is public. Is that intentional?

**Comment by grennis** [comment]
My mistake, I see now that this PR is obsolete and code has been replaced. My confusion was over the availability of one of create/update but not the other, but I commented separately about that.

**Pull request closed without merging**